ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,305 @@
+            args=("-s", "--storage-sibling"),
+            metavar="STORAGE",
+            doc="""name of the RIA storage sibling. Must not be identical to SIBLING. By default SIBLING is appended 
+            with '-storage'""",
+            constraints=EnsureStr() | EnsureNone()),
+        force=Parameter(
+            args=("-f", "--force"),
+            doc="""don't fail on existing siblings. Use and possibly reconfigure them instead.""",
+            action='store_true'),
+        no_publish=Parameter(
+            args=("--no-publish",),
+            doc="""whether to publish the dataset's history (no data) to SIBLING after creation.""",
+            action="store_true"),
+        no_server=Parameter(
+            args=("--no-server",),
+            doc="""don't assume the store to be served by a webserver. By default git's update-server-info hook is 
+            enabled. With this option that can be disabled.""",
+            action="store_true"),
+        recursive=recursion_flag,
+    )
+
+    @staticmethod
+    @datasetmethod(name='create_sibling_ria')
+    @eval_results
+    def __call__(
+            sibling,
+            dataset=None,
+            storage_sibling=None,
+            force=False,
+            no_publish=False,
+            no_server=False,
+            recursive=False,
+    ):
+
+        # TODO: is check_installed actually required?
+        ds = require_dataset(dataset, check_installed=True, purpose='create sibling RIA')
+        res_kwargs = dict(
+            ds=ds,
+            action="create-sibling-ria",
+            logger=lgr,
+        )
+
+        if ds.repo.get_hexsha() is None or ds.id is None:
+            raise RuntimeError(
+                "Repository at {} is not a DataLad dataset, "
+                "run 'datalad create' first.".format(ds.path))
+
+        if not storage_sibling:
+            storage_sibling = "{}-storage".format(sibling)
+
+        if sibling == storage_sibling:
+            # leads to unresolvable, circular dependency with publish-depends
+            raise ValueError("sibling names must not be equal")
+
+        # TODO: messages - this is "create-sibling". Don't confuse existence of local remotes with existence of the
+        #       actual remote sibling in wording
+        if not force and sibling in [r['name'] for r in ds.siblings(result_renderer=None)]:
+            yield get_status_dict(
+                status='error',
+                message="a sibling '{}' is already configured. Use --force to overwrite it.".format(sibling),
+                **res_kwargs,
+            )
+            return
+
+        if not force and storage_sibling in [r['name'] for r in ds.siblings(result_renderer=None)]:
+            yield get_status_dict(
+                status='error',
+                message="a storage-sibling '{}' is already configured. Use --force to overwrite it.".format(storage_sibling),
+                **res_kwargs,
+            )
+            return
```

Comment:
The above calls `siblings()` twice and pulls out the exact same info. The call could be relatively expensive, depending on the setup.